subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2006,12 +2008,24 @@ int lwm2m_engine_update_observer_max_period(char *pathstr, uint32_t period_s)
 }
 
 void lwm2m_engine_get_binding(char *binding)
+{
+	/* Defaults to UDP. */
+	strncpy(binding, "U", BINDING_OPT_MAX_LEN);
```

Comment:
Can we guarantee that binding does not overflow i.e., should the max length be given as a parameter to this func and then check it too? If we know that the binding has always space for it, then I think we probably do not need to check that.

Reply:
This is internal function to engine, and it has been without size parameter so far.
The binding letters are defined in LwM2M standard, and I would therefore object against delivering a size of string that is known.
If we would have Doxygen type of comment blocks in `lwm2m_engine.h` I could add the remark there that the buffer should have sufficien space to hold  a binding string "UQ". But, none of the functions there are commented, so I did not add it.
If anything, we should probably refactor binding to be enum and queue mode to be a boolean flag.